Updates the jukebox, including new sprites, descriptions, and music tracks.

Sunbeamstress opened the pull request:
🆑 
tweak: The jukebox has been resprited.
add: New tracks added to the jukebox music selection; some have had their names corrected.
/ 🆑 

- Changes the sprite from some old goofy retro 50s diner box to a wide touchscreen panel with animated interface.
- Refluffs it into a "retro jukebox" by the Auraliving company, that is still higher-tech than what we had.
- Dumps a bunch of the game's musical tracks into the jukebox selection.

Base sprite:
![](https://i.imgur.com/Z4MWhKg.gif)

Playing:
![](https://i.imgur.com/HEpkEvU.gif)

Broken:
![](https://i.imgur.com/2kUMfKn.gif)

Emagged:
![](https://i.imgur.com/o9k9EUp.gif)

In-game:
![](https://i.imgur.com/7Cw49jQ.png)

Sunbeamstress committed:
[ SQUASH! ]

Track title change. Moved around the bar a tiny bit to add the new jukebox. Preserves the old one.

Adds an "old" jukebox variant with the old style.

Updates the jukebox, and adds extra music selections.

Sunbeamstress commented:
Ack, my bad! I forgot to update the example sprites before I fattened up the base. It looks like this, for reference:

![](https://i.imgur.com/UJPY4Hz.png)

Alex6511 commented:
I think you should leave the old sprite/jukebox in for posterity, personally.

Crushtoe commented:
Eh. Looks more karaoke than jukebox.

Sunbeamstress commented:
@Alex6511 It's still there! The new sprite uses jukebox_new.dmi.

babydoll commented:
I think you should keep the one in the dead lounge as the old art, and add this somewhere proper. It's time to add a freely and easily accessible jukebox. I think we're ready.

e: Also, not that I have much room to talk to someone who actually makes decent icons, have you tried adding some slightly transparent pixels to these edges here? Might make it a little less harsh

![](https://i.imgur.com/y8pRgf0.png)

PsiOmegaDelta commented:
It's pretty trivial to have both with some subtyping.

PsiOmegaDelta commented on code/game/machinery/jukebox.dm:
```diff
@@ -31,16 +36,35 @@ datum/track/New(var/title_name, var/audio)
 		new/datum/track("Clouds of Fire", 'sound/music/clouds.s3m'),
-		new/datum/track("D`Bert", 'sound/music/title2.ogg'),
-		new/datum/track("D`Fort", 'sound/ambience/song_game.ogg'),
+		new/datum/track("Dehumanize Yourself and Face to Bloodshed", 'sound/music/title2.ogg'),
```
Technically named dilbert.mp3 from the "album" Dehumanize Yourself and Face to Bloodshed, based of robocop.mp3
Mostly noting where the name choice came from.

PsiOmegaDelta commented on code/game/machinery/jukebox.dm:
```diff
@@ -31,16 +36,35 @@ datum/track/New(var/title_name, var/audio)
-		new/datum/track("D`Bert", 'sound/music/title2.ogg'),
-		new/datum/track("D`Fort", 'sound/ambience/song_game.ogg'),
+		new/datum/track("Dehumanize Yourself and Face to Bloodshed", 'sound/music/title2.ogg'),
+		new/datum/track("Asteroids", 'sound/ambience/song_game.ogg'),
```
This track comes from Dwarf Fortress, hence the original naming.

chaoko99 replied:
Pretty sure it's not even supposed to be in this codebase. Is DF GPL?

Sunbeamstress commented:
@PsiOmegaDelta the rationale there was to make them a little more lore friendly.  I'm waging a one-man war on memes in the repo.

@sabiram you don't have to be an artist to recognize flaws in art!  I agree,  and will look into softening the corners of the panel when I get home.

Sunbeamstress commented:
PS. My intention is to leave the classic jukebox in the abandoned bar (/jukebox/old), and put a mediatron in the mess. We have a shitload of map changes in open PRs right now,  hence not doing it immediately

Crushtoe commented:
Fair enough, then. I like.

PsiOmegaDelta commented:
You're waging a war against memes yet introduce "Dehumanize Yourself and Face to Bloodshed"?

Sunbeamstress commented:
Lol, give me a name that isn't dilbert.mp3 and I'll call it whatever you want

chaoko99 commented:
WELL TECHNICALLY, it's called stage3

Sunbeamstress commented:
Alright!

@sabiram: it goes against every rule of pixel art I hold dear, but making some of the jagged corners slightly transparent really helps it blend and communicate the "curve" of the screen.

![](https://i.imgur.com/MfiESDM.png)

Deck Four is once again hiding an old-school jukebox, so you can relive the glory days of the American 1950s. The new one lives in the bar. I removed a few stools and tables and moved some stuff into the corner to make room for it.

![](https://i.imgur.com/3xmm6VB.png)

And a shot of what the music selection looks like now.

![](https://i.imgur.com/3GsGogH.png)

chaoko99 commented:
>![image](https://user-images.githubusercontent.com/4706052/37941175-b547e00e-3121-11e8-8872-cc83c1012f73.png)

No.  I seriously am against the moving of those vending machines. That spot is used rather regularly by people, you know.

Besides, you are aware that this makes that thing's screen two meters long, right? That means it's a livingroom length TV. That doesn't seem at all out of place to you in a cafeteria, right? Right?

Furthermore, as much as I like the sprite... It clashes horribly with the rest of the game. It's too wide, it's too colorful, it's too black, and it's certainly just too large. It'll also fight with anything that is adjacent to it due to how byond's layering works. Firm 👎 from me. I like the concept, I don't like the execution.

All this does is make it easier to force our horrendous soundtrack down my ears via a buggy as shit jukebox that functions on line of sight rather than actual sound cues.

mustafakalash commented:
Other things that are two meters long: crowbars, books, ID cards, single bills

chaoko99 commented:
None of those things have ` density = TRUE`

Also, none of those take up 64x32. Only 32x32.

Sunbeamstress commented:
Technically the screen is 48x32.

chaoko99 commented:
So it's a meter and a half. Still very large and out of place. Make it a wall mount?

Seriously... Nobody is going to floormount something that large.

Sunbeamstress commented:
![](http://4k.com/wp-content/uploads/2015/12/398146.jpg)

??????

chaoko99 commented:
You put those on tables.

ITT: Earthcrusher does not Feng shui

chaoko99 commented:
https://i.imgur.com/VqSDIjR.png Put it right here, on the wall. Pixel shift it. Just don't edit that section from what it was.

Sunbeamstress commented:
I'll uh, see if anyone else has objections, thanks.

comma reviewed (approved):
Eh either is fine
